recipes/egl/system/conanfile.py

```diff
@@ -16,54 +18,24 @@ def configure(self):
         if self.settings.os not in ["Linux", "FreeBSD"]:
             raise ConanInvalidConfiguration("This recipes supports only Linux and FreeBSD")
             
+    # TODO: check how to do with system packages in Conan 2.0
     def package_id(self):
         self.info.header_only()
 
-    def _fill_cppinfo_from_pkgconfig(self, name):
-        pkg_config = tools.PkgConfig(name)
-        if not pkg_config.provides:
-            raise ConanException("EGL development files aren't available, give up")
-        libs = [lib[2:] for lib in pkg_config.libs_only_l]
-        lib_dirs = [lib[2:] for lib in pkg_config.libs_only_L]
-        ldflags = [flag for flag in pkg_config.libs_only_other]
-        include_dirs = [include[2:] for include in pkg_config.cflags_only_I]
-        cflags = [flag for flag in pkg_config.cflags_only_other if not flag.startswith("-D")]
-        defines = [flag[2:] for flag in pkg_config.cflags_only_other if flag.startswith("-D")]
+    def system_requirements(self):
+        yum = Yum(self).install(["mesa-libEGL-devel"])
+        try:
+            # this works for ubuntu>=20, debian>=11, pop>=20
+            apt = Apt(self).install(["libegl-dev"])
+        except ConanException:
+            apt = Apt(self).install(["libegl1-mesa-dev"])
```

Comment:
TODO: Find some syntax to do this better than chaining try: except:

Reply:
We will probably implement something like:
`apt = Apt(self).install_alternatives(["libegl-dev"], ["libegl1-mesa-dev"])`

Reply:
Some python experts prefer only try..except instead of a condition. So it's good enough